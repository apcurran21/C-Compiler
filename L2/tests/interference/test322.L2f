(@main
	0
	rdi <- 7
	rsi <- 1
	call allocate 2
	%jennysNumber <- rax
	%tempJenny <- %jennysNumber
	rdi <- 8
	mem rsp -8 <- :call_label0
	call @Jenny 1
	:call_label0
	%retVal <- rax
	mem %tempJenny 0 <- %retVal
	%tempJenny += 8
	rdi <- 6
	mem rsp -8 <- :call_label1
	call @Jenny 1
	:call_label1
	%retVal <- rax
	mem %tempJenny 0 <- %retVal
	%tempJenny += 8
	rdi <- 7
	mem rsp -8 <- :call_label2
	call @whoCanI 1
	:call_label2
	%retVal <- rax
	mem %tempJenny 0 <- %retVal
	%tempJenny += 8
	rdi <- 5
	mem rsp -8 <- :call_label3
	call @TurnTo 1
	:call_label3
	%retVal <- rax
	mem %tempJenny 0 <- %retVal
	%tempJenny += 8
	rdi <- 3
	mem rsp -8 <- :call_label4
	call @youGiveMeSomething 1
	:call_label4
	%retVal <- rax
	mem %tempJenny 0 <- %retVal
	%tempJenny += 8
	rdi <- 0
	mem rsp -8 <- :call_label5
	call @iCanHold 1
	:call_label5
	%retVal <- rax
	mem %tempJenny 0 <- %retVal
	%tempJenny += 8
	rdi <- 9
	mem rsp -8 <- :call_label6
	call @onTo 1
	:call_label6
	%retVal <- rax
	mem %tempJenny 0 <- %retVal
	rdi <- %jennysNumber
	mem rsp -8 <- :call_label7
	call @gimmeACall 1
	:call_label7
	return
)
