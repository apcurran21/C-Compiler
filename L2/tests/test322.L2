(@main
  (@main
    0
    rdi <- 15
    rsi <- 5
    rdx <- 0
    rcx <- 10
    r8 <- -25
    r9 <- 90
    mem rsp -24 <- 2
    mem rsp -16 <- 1

mem rsp -8 <- :call_label0
    call @max 8

:call_label0
    %ret <- rax
    %ret <- %ret
    %ret <<= 1
    %ret <- %ret
    %ret += 1
    rdi <- %ret
    call print 1
    rdi <- -13
    rsi <- 6
    rdx <- 77
    rcx <- 15
    r8 <- -2
    r9 <- 54
    mem rsp -24 <- 1
    mem rsp -16 <- 0

mem rsp -8 <- :call_label1
    call @max 8

:call_label1
    %ret <- rax
    %ret <- %ret
    %ret <<= 1
    %ret <- %ret
    %ret += 1
    rdi <- %ret
    call print 1
    return
  )
  (@max
    8
    %v1 <- rdi
    %v2 <- rsi
    %v3 <- rdx
    %v4 <- rcx
    %v5 <- r8
    %v6 <- r9
    %v7 <- stack-arg 0
    %v8 <- stack-arg 8
    %test <- %v2 < %v1

    cjump %test = 1 :call_label_ciao_2
    %v1 <- %v2
    :call_label_ciao_2
    %test <- %v3 < %v1

    cjump %test = 1 :call_label_ciao_3
    %v1 <- %v3
    :call_label_ciao_3
    %test <- %v4 < %v1

    cjump %test = 1 :call_label_ciao_4
    %v1 <- %v4
    :call_label_ciao_4
    %test <- %v5 < %v1

    cjump %test = 1 :call_label_ciao_5
    %v1 <- %v5
    :call_label_ciao_5
    %test <- %v6 < %v1

    cjump %test = 1 :call_label_ciao_6
    %v1 <- %v6
    :call_label_ciao_6
    %test <- %v7 < %v1

    cjump %test = 1 :call_label_ciao_7
    %v1 <- %v7
    :call_label_ciao_7
    %test <- %v8 < %v1

    cjump %test = 1 :call_label_ciao_8
    %v1 <- %v8
    :call_label_ciao_8
    rax <- %v1
    return
  )
)
