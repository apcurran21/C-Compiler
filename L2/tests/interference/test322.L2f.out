%jennysNumber r10 r11 r12 r13 r14 r15 r8 r9 rax rbp rbx rcx rdi rdx %retVal rsi %tempJenny
r10 %jennysNumber r11 r12 r13 r14 r15 r8 r9 rax rbp rbx rcx rdi rdx rsi %tempJenny
r11 %jennysNumber r10 r12 r13 r14 r15 r8 r9 rax rbp rbx rcx rdi rdx rsi %tempJenny
r12 %jennysNumber r10 r11 r13 r14 r15 r8 r9 rax rbp rbx rcx rdi rdx %retVal rsi %tempJenny
r13 %jennysNumber r10 r11 r12 r14 r15 r8 r9 rax rbp rbx rcx rdi rdx %retVal rsi %tempJenny
r14 %jennysNumber r10 r11 r12 r13 r15 r8 r9 rax rbp rbx rcx rdi rdx %retVal rsi %tempJenny
r15 %jennysNumber r10 r11 r12 r13 r14 r8 r9 rax rbp rbx rcx rdi rdx %retVal rsi %tempJenny
r8 %jennysNumber r10 r11 r12 r13 r14 r15 r9 rax rbp rbx rcx rdi rdx rsi %tempJenny
r9 %jennysNumber r10 r11 r12 r13 r14 r15 r8 rax rbp rbx rcx rdi rdx rsi %tempJenny
rax %jennysNumber r10 r11 r12 r13 r14 r15 r8 r9 rbp rbx rcx rdi rdx rsi %tempJenny
rbp %jennysNumber r10 r11 r12 r13 r14 r15 r8 r9 rax rbx rcx rdi rdx %retVal rsi %tempJenny
rbx %jennysNumber r10 r11 r12 r13 r14 r15 r8 r9 rax rbp rcx rdi rdx %retVal rsi %tempJenny
rcx %jennysNumber r10 r11 r12 r13 r14 r15 r8 r9 rax rbp rbx rdi rdx rsi %tempJenny
rdi %jennysNumber r10 r11 r12 r13 r14 r15 r8 r9 rax rbp rbx rcx rdx rsi %tempJenny
rdx %jennysNumber r10 r11 r12 r13 r14 r15 r8 r9 rax rbp rbx rcx rdi rsi %tempJenny
%retVal %jennysNumber r12 r13 r14 r15 rbp rbx %tempJenny
rsi %jennysNumber r10 r11 r12 r13 r14 r15 r8 r9 rax rbp rbx rcx rdi rdx %tempJenny
%tempJenny %jennysNumber r10 r11 r12 r13 r14 r15 r8 r9 rax rbp rbx rcx rdi rdx %retVal rsi
